src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java

```diff
@@ -223,8 +223,11 @@
 
         // TODO reading persistent identify immediately forces the lazy load of the preferences, and defeats the
         // purpose of PersistentIdentity's laziness.
-        final String peopleId = mPersistentIdentity.getPeopleDistinctId();
-        mDecideMessages.setDistinctId(peopleId);
+        String decideId = mPersistentIdentity.getPeopleDistinctId();
+        if (null == decideId) {
```

Comment:
I don't think I fully understand what's going on here.. Why do you use events distinct id as the decideId? Doesn't it change the behavior of this whole thing?